### PR TITLE
feat(content-uploader): fix folder upload failure display and add a resume callback

### PR DESCRIPTION
**Pull request by borajimin**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @borajimin, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: src/elements/content-uploader/ContentUploader.js [hunk 1]
```diff
@@ -71,6 +71,7 @@ type Props = {
     onComplete: Function,
     onError: Function,
     onMinimize?: Function,
+    onResume: Function,
```
Comment (mickr) [inline src/elements/content-uploader/ContentUploader.js]: This should probably be added to the documentation in the readme

File: src/elements/content-uploader/ContentUploader.js [hunk 2]
```diff
@@ -849,7 +852,10 @@ class ContentUploader extends Component<Props, State> {
         const someUploadHasFailed = items.some(uploadItem => uploadItem.status === STATUS_ERROR);
         const allItemsArePending = !items.some(uploadItem => uploadItem.status !== STATUS_PENDING);
         const noFileIsPendingOrInProgress = items.every(
-            uploadItem => uploadItem.status !== STATUS_PENDING && uploadItem.status !== STATUS_IN_PROGRESS,
+            uploadItem =>
+                (uploadItem.status === STATUS_COMPLETE || uploadItem.status === STATUS_ERROR) &&
+                uploadItem.status !== STATUS_PENDING &&
+                uploadItem.status !== STATUS_IN_PROGRESS,
```
Comment (mickr) [inline src/elements/content-uploader/ContentUploader.js]: Is there a test that should be updated to accommodate this change?

Comment (borajimin) [reply src/elements/content-uploader/ContentUploader.js]: I don't think this change affects the function's behavior significantly enough to add a test. Line 856 was added to ensure the upload status is either successful or failed upon completion. There is a test that covers updateViewAndCollection; however, it was not affected by the change and a lot of the functionalities are not covered in the test suite. Please share your thoughts if you feel strongly about the adding a test.

File: src/elements/content-uploader/ContentUploader.js (hunk 1 above)
Comment (borajimin) [reply src/elements/content-uploader/ContentUploader.js]: 👍

**Pull request merged**